src/main/java/htsjdk/samtools/util/Writer.java

```diff
@@ -0,0 +1,17 @@
+package htsjdk.samtools.util;
+
+import java.io.Closeable;
+import java.io.Flushable;
+
+
+public interface Writer<A> extends Closeable, Flushable {
```

Comment:
I'm curious why extend `Flushable`?  Is it really necessary above to call `flush()` after writing a batch out?  What makes us think we know better when to flush() than the underlying writer?

Reply:
Fixed, this ended up being an issue in how I had integrated into Picard, and not the Writer. So nothing implements Flushable anymore.